src/fairchem/data/oc/utils/geometry.py

```diff
@@ -0,0 +1,264 @@
+
+        plane1 = Geometry.vec_and_point_to_plane(n1, origin)
+        plane2 = Geometry.vec_and_point_to_plane(n2, origin)
+        plane3 = Geometry.vec_and_point_to_plane(n3, origin)
+        plane4 = Geometry.vec_and_point_to_plane(-n1, top)
+        plane5 = Geometry.vec_and_point_to_plane(-n2, top)
+        plane6 = Geometry.vec_and_point_to_plane(-n3, top)
+
+        return [plane1, plane2, plane3, plane4, plane5, plane6]
+
+    @staticmethod
+    def extract_box_properties(center, length, lo_corner, hi_corner):
+        """Given two of the properties 'center', 'length', 'lo_corner',
+        'hi_corner', return all the properties. The properties that
+        are not given are expected to be 'None'.
+        """
+        # exactly two arguments have to be non-none
+        if sum(x is None for x in [center, length, lo_corner, hi_corner]) != 2:
+            raise ValueError("Exactly two arguments have to be given")
+
+        # declare arrays to allow mathematical operations
+        center, length = np.asarray(center), np.asarray(length)
+        lo_corner, hi_corner = np.asarray(lo_corner), np.asarray(hi_corner)
+        relations = [
+            [
+                "lo_corner",
+                "hi_corner - length",
+                "center - length / 2",
+                "2 * center - hi_corner",
+            ],
+            [
+                "hi_corner",
+                "lo_corner + length",
+                "center + length / 2",
+                "2 * center - lo_corner",
+            ],
+            [
+                "length / 2",
+                "(hi_corner - lo_corner) / 2",
+                "hi_corner - center",
+                "center - lo_corner",
+            ],
+            [
+                "center",
+                "(hi_corner + lo_corner) / 2",
+                "hi_corner - length / 2",
+                "lo_corner + length / 2",
+            ],
+        ]
+
+        # compute all relations
+        relation_list = []
+        for relation in relations:
+            for i in relation:
+                try:
+                    relation_list.append(eval(i))
+                except TypeError:
+                    continue
+
+        # keep the non-None relations
+        for i, relation in enumerate(relation_list):
+            if None in relation:
+                del relation_list[i]
+        return relation_list
+
+    def packmol_structure(self, filename, number, side):
+        """Make structure to be used in PACKMOL input script
+
+        :param number: number of solvent molecules
+        :type number: int
+        :param side: pack solvent inside/outside of geometry
+        :type side: str
+        :returns: string with information about the structure
+        :rtype: str
+        """
+        structure = ""
+        structure += f"structure {filename}\n"
+        structure += f"  number {number}\n"
+        structure += f"  {side} {self.__repr__()} "
+        for param in self.params:
```

Comment:
I don't see self.params defined for `Geometry`. Is the `Geometry` supposed to be an abstract base class?
If so, I would explicitly have it derive from `abc.ABC` , make `packmol_structure` an abstractmethod here, and have this definition of the `packmol_structure` method go in `BoxGeometry`? I assume thats the intended use?

Reply:
`geometry.py` hasn't really been modified from its original implementation - https://github.com/henriasv/molecular-builder/blob/master/molecular_builder/geometry.py.

I took it as is and didn't want to introduce it as a dependency. But I can clean this up a little.